builtin/providers/aws/resource_aws_db_event_subscription.go

```
@@ -0,0 +1,270 @@
+	}
+
+	eventCategoriesSet := d.Get("event_categories").(*schema.Set)
+	eventCategories := make([]*string, eventCategoriesSet.Len())
+	for i, eventCategory := range eventCategoriesSet.List() {
+		eventCategories[i] = aws.String(eventCategory.(string))
+	}
+
+	request := &rds.CreateEventSubscriptionInput{
+		SubscriptionName: aws.String(name),
+		SnsTopicArn:      aws.String(d.Get("sns_topic").(string)),
+		Enabled:          aws.Bool(d.Get("enabled").(bool)),
+		SourceIds:        sourceIds,
+		SourceType:       aws.String(d.Get("source_type").(string)),
+		EventCategories:  eventCategories,
+		Tags:             tags,
+	}
+
+	log.Println("[DEBUG] Create RDS Event Subscription:", request)
+
+	_, err := rdsconn.CreateEventSubscription(request)
+	if err != nil {
+		return fmt.Errorf("Error creating RDS Event Subscription %s: %s", name, err)
+	}
+
+	log.Println(
+		"[INFO] Waiting for RDS Event Subscription to be ready")
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"creating"},
+		Target:     []string{"active"},
+		Refresh:    resourceAwsDbEventSubscriptionRefreshFunc(d, meta),
+		Timeout:    40 * time.Minute,
+		MinTimeout: 10 * time.Second,
+		Delay:      30 * time.Second, // Wait 30 secs before starting
+	}
+
+	// Wait, catching any errors
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return err
+	}
+
+	return resourceAwsDbEventSubscriptionRead(d, meta)
+}
+
+func resourceAwsDbEventSubscriptionRead(d *schema.ResourceData, meta interface{}) error {
+	sub, err := resourceAwsDbEventSubscriptionRetrieve(d, meta)
+	if err != nil {
+		return err
+	}
+	if sub == nil {
+		d.SetId("")
+		return nil
+	}
+
+	d.SetId(*sub.CustSubscriptionId)
+	if err := d.Set("name", sub.CustSubscriptionId); err != nil {
+		return err
+	}
+	if err := d.Set("sns_topic", sub.SnsTopicArn); err != nil {
+		return err
+	}
+	if err := d.Set("source_type", sub.SourceType); err != nil {
+		return err
+	}
+	if err := d.Set("enabled", sub.Enabled); err != nil {
```

Typically we ignore errors from most `d.Set` calls unless these are setting non-primitive data structures (i.e. anything except `TypeInt`, `TypeString`, `TypeFloat` and `TypeBool`), but these checks do no harm. 👍